   Adds a configration to select whether to add -funit-at-a-time to
   compiler option.

---
 Makefile     |    2 	2 +	0 -	0 !
 init/Kconfig |   12 	12 +	0 -	0 !
 2 files changed, 14 insertions(+)

Index: linux-2.6.23-rc9/init/Kconfig
===================================================================
--- linux-2.6.23-rc9.orig/init/Kconfig	2007-10-08 23:51:25.000000000 +0200
+++ linux-2.6.23-rc9/init/Kconfig	2007-10-09 00:30:58.000000000 +0200
@@ -571,6 +571,18 @@
 
 endchoice
 
+config CC_FUNIT_AT_A_TIME
+       bool "add -funit-at-a-time to gcc option" if EMBEDDED
+       default n
+       help
+	    Enabling this option will pass "-funit-at-a-time" to gcc.
+	    What it does is to first load the whole file into memory
+	    and then generate the output. This allows it to use a
+	    better inlining strategy, drop unused static functions and
+	    use -mregparm automatically for static functions.
+	    Some tests show impressive .text shrinkage from using
+	    unit-at-a-time compilation.
+
 endmenu		# General setup
 
 config RT_MUTEXES
Index: linux-2.6.23-rc9/Makefile
===================================================================
--- linux-2.6.23-rc9.orig/Makefile	2007-10-08 23:52:27.000000000 +0200
+++ linux-2.6.23-rc9/Makefile	2007-10-08 23:52:27.000000000 +0200
@@ -510,7 +510,9 @@
 
 # Enable unit-at-a-time mode when possible. It shrinks the
 # kernel considerably.
+ifdef CONFIG_CC_FUNIT_AT_A_TIME
 CFLAGS += $(call cc-option,-funit-at-a-time,)
+endif
 
 # warn about C99 declaration after statement
 CFLAGS += $(call cc-option,-Wdeclaration-after-statement,)
